src/compiler/program.ts

```diff
@@ -4795,7 +4796,7 @@ export function createProgram(rootNamesOrOptions: readonly string[] | CreateProg
     }
 
     function verifyProjectReferences() {
-        const buildInfoPath = !options.suppressOutputPathCheck ? getTsBuildInfoEmitOutputFilePath(options) : undefined;
+        const buildInfoPath = !some(internalOptionDeclarations, d => hasProperty(options, d.name)) ? getTsBuildInfoEmitOutputFilePath(options) : undefined;

```

Comment:
I dont think this is correct change. The intention here to get buildInfo path is to check if the paths collide with any of referenced project's buildInfo and hence we were checking `suppressOutputPathCheck` I dont think this should be disabled if any of the internal options are true?

Reply:
Ah, yeah, I see, you were talking about putting a check like this *inside* `getTsBuildInfoEmitOutputFilePath` - one sec...

Reply:
There we go, moved. Still, it feels weird to treat internal options any differently than public ones, except, perhaps that we haven't taken the time to figure out what levels of incremental invalidation need to be set in their option description object yet.